File: 2025/day06_1/src/main/kotlin/CephalopodOperation.kt
package org.tlarsen.adventofcode2025

interface CephalopodCalculation {
    fun calculate(leftVal: Long, rightVal: Long): Long
}

enum class CephalopodOperation : CephalopodCalculation {
    ADDITION {
        override fun calculate(leftVal: Long, rightVal: Long): Long {
            return leftVal + rightVal
        }

        override fun toString(): String {
            return "+"
        }
    },
    MULTIPLICATION {
        override fun calculate(leftVal: Long, rightVal: Long): Long {
            return leftVal * rightVal
        }

        override fun toString(): String {
            return "*"
        }
    },
    NONE {
        override fun calculate(leftVal: Long, rightVal: Long): Long {
            return 0L
        }

        override fun toString(): String {
            return "(none)"
        }
    }
}

File: 2025/day06_1/src/main/python/cephalopod_operation.py
from enum import Enum


class CephalopodOperation(Enum):
    ADDITION = "+"
    MULTIPLICATION = "*"
    NONE = "(none)"

    def calculate(self, left_val: int, right_val: int) -> int:
        if self == CephalopodOperation.ADDITION:
            return left_val + right_val
        elif self == CephalopodOperation.MULTIPLICATION:
            return left_val * right_val
        else:
            return 0